src/ui/App.tsx

```diff
@@ -67,16 +73,35 @@ function App() {
     new Set(),
   );
   const [showCheckboxes, setShowCheckboxes] = useState(false);
+  const previousPackages = useRef<Set<string>>(new Set());
 
   useEffect(() => {
-    fetch('/api/packages')
+    const majorBumps = Object.entries(selections)
+      .filter(([_, type]) => type === 'major')
+      .map(([pkgName]) => pkgName);
+
+    fetch(`/api/packages?majorBumps=${majorBumps.join(',')}`)
       .then((res) => {
         if (!res.ok) {
           throw new Error(`Received ${res.status}`);
         }
         return res.json();
       })
       .then((data: Package[]) => {
+        const newPackageNames = new Set(data.map((pkg) => pkg.name));
+
+        // Only clean up selections if the package list actually changed
+        if (!setsAreEqual(previousPackages.current, newPackageNames)) {
+          setSelections((prev) =>
```

Comment:
Ah, I think this name is confusing me a bit. At first I thought this referred to the packages that users had selected via the checkbox. But this refers to packages that are shown in the list, is that true?

It's probably fine now, but perhaps we can consider changing this name in a future PR.

Reply:
I did change it to `releaseSelections`

Reply:
Hmm, I think we should do this in a new PR since it touches a bunch of new places. There are already a couple of distinct changes in this PR and maybe we shouldn't try to add more things.

Reply:
Yes make sense, I did rollback the changes